WordPress/gutenberg
# Add preferences modal to site editor

## What?


This PR adds the Preferences modal to the site editor, with the aim of later filling in all the preferences we might find useful.

## Why?


Split off from [this comment](https://github.com/WordPress/gutenberg/pull/38317#issuecomment-1023961561). Tl,dr: in order to add more user preferences common to all editors, it's best to show them in a common UI. 

Related: #31965

## How?


Adds the recently abstracted reusable PreferencesModal from the interface package. For now, this adds only the "Spotlight mode" preference umder "General" and the "Contain text cursor inside block" preference under "Blocks". The plan is to progressively add all the other settings that make sense in the site editor, such as "Display button labels" (in progress: #38317) and the "Visible blocks" section. Suggestions welcome as to which other options we might add 🙂 

## Testing Instructions





1. Open the Site Editor;
2. Click the top right hand "Options" button in the header;
3. Check that there is a "Preferences" button as the last item of the resulting dropdown;
4. Check that clicking "Preferences" opens a modal with two tabs, and one preference under each.

## Screenshots 

<img width="1103" alt="" src="https://user-images.githubusercontent.com/8096000/158728917-1f925a89-d03b-40e2-a898-527d60bf6c9f.png">
